docs/spec/distributing.rst

```diff
@@ -10,72 +10,428 @@ Stub files
+"""""""""""""""""""""""
+
+Module level variables and constants can be annotated using either
+type comments or variable annotation syntax::
+
+    x: int  # recommended
+    x: int = 0
+    x = 0  # type: int
+    x = ...  # type: int
+
+The ellipsis literal can stand in for any value::
+
+    x: int = ...  # type is int
+
+A variable annotated as ``Final`` and assigned a literal value has the
+corresponding ``Literal`` type::
+
+    x: Final = 0  # type is Literal[0]
+
+In all other cases, the type of a variable is unspecified when the variable is
+unannotated or when the annotation and the assigned value disagree::
+
+    x = 0  # type is unspecified
+    x = ...  # type is unspecified
+    x: int = ""  # type is unspecified
+
+Classes
+"""""""
+
+Class definition syntax follows general Python syntax, but type checkers
+are only expected to understand the following constructs in class bodies:
+
+* The ellipsis literal ``...`` is ignored and used for empty
+  class bodies. Using ``pass`` in class bodies is undefined.
+* Instance attributes follow the same rules as module level attributes
+  (see above).
+* Method definitions (see below) and properties.
+* Method aliases.
+* Inner class definitions.
+
+Yes::
+
+    class Simple: ...
+
+    class Complex(Base):
+        read_write: int
+        @property
+        def read_only(self) -> int: ...
+        def do_stuff(self, y: str) -> None: ...
+        doStuff = do_stuff
+        class Inner: ...
+
+More complex statements don't need to be supported.
+
+The type of generic classes can be narrowed by annotating the ``self``
+argument of the ``__init__`` method::
+
+    class Foo(Generic[_T]):
+        @overload
+        def __init__(self: Foo[str], type: Literal["s"]) -> None: ...
+        @overload
+        def __init__(self: Foo[int], type: Literal["i"]) -> None: ...
+        @overload
+        def __init__(self, type: str) -> None: ...
+
+The class must match the class in which it is declared. Using other classes,
+including sub or super classes, will not work. In addition, the ``self``
+annotation cannot contain type variables.
+
+Functions and Methods
+"""""""""""""""""""""
 
-* Modules that must be compatible with Python 2 and 3
+Function and method definition syntax follows general Python syntax.
+For backwards compatibility, positional-only parameters can also be marked by
+prefixing their name with two underscores (but not suffixing it with two
+underscores)::
 
-* Modules that use annotations for other purposes
+    # x is positional-only
+    # y can be used positionally or as keyword argument
+    # z is keyword-only
+    def foo(x, /, y, *, z): ...  # recommended
+    def foo(__x, y, *, z): ...  # backwards compatible syntax
 
-Stub files have the same syntax as regular Python modules.  There is one
-feature of the ``typing`` module that is different in stub files:
-the ``@overload`` decorator described below.
+If an argument or return type is unannotated, per :pep:`484` its
```

Comment:
The typing spec shouldn't be quoting PEP 484. It should refer to itself.

In this case, the wording for return types has been updated since PEP 484 to allow for return type inference, so this section should be careful not to contradict other sections of the typing spec.

Reply:
Removed this sentence, since it didn't have anything stub-specific to say.